### DISCUSSION TITLE
cast storage does not work for a live contract

### DISCUSSION THREAD

**Discussion by cristiansoto-f** [Cyfrin/foundry-full-course-cu#1286] [General]
Hey

I get the following error when I try to look at USDC storage on sepolia:
```
cast storage --rpc-url $SEPOLIA_RPC_URL 0xf08A50178dfcDe18524640EA6618a1f965821715
[⠆] Compiling...
No files changed, compilation skipped
No matching artifacts found, fetching source code from Etherscan...
Error: 
You must provide an Etherscan API key if you're fetching a remote contract's storage.
```
What I am doing wrong?

**Comment by ghost** [comment]
The error states there is an issue with your Etherscan API Key.

Things to consider:

1. Did you added your sepolia rpc url in the `.env` file with `SEPOLIA_RPC_URL` as the name?
2. Did you run `source` to load the enviroment variables before running `cast storage`?

**Comment by xyz899** [comment]
### Hey,
#### When you want to access a particular contract indexed on a testnet or mainnet you will need to provide an Etherscan API Key, to create and get a fresh one, you have to go directly on [Etherscan](https://etherscan.io/) , create a new account and get a new API Key ( steps are almost the same as getting an RPC URL)

Now add your API KEY in your `.env` where your RPC-URL is : 
```
ETHERSCAN_API_KEY=YourKey>
SEPOLIA_RPC_URL=<YourRPCURL>
```

Then don't forget to make your environment variables available by typing `source .env`

and lastly , your Command Line would look like this :
```
cast storage --rpc-url $SEPOLIA_RPC_URL 0xf08A50178dfcDe18524640EA6618a1f965821715  --etherscan-api-key <YourKey or $ETHERSCAN_API_KEY>
```
 (alternatively you can use `-e` instead of the full `--etherscan-api-key`)

Hope that helps you !

**Comment by cristiansoto-f** [comment]
Thank you both of you.

It works but only on Ethereum MAINNET and not on SEPOLIA (Error: Invalid API Key)

```cast storage --rpc-url $MAINNET_RPC_URL -e $ETHERSCAN_API_KEY 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48```

do you know if there's a way to cast storage on sepolia chain?
